### PR TITLE
ARC-1416 tag backfill requests

### PR DESCRIPTION
**Pull request by tavomaciel**
**What's in this PR?**

Tags requests with `operationType: "NORMAL"` or `operationType: "BACKFILL"`

**Why**

This helps real-time data get available faster, as it's not fighting for resources with backfill data.
Jira might also soon start to rate limit NORMAL requests.

**Added feature flags**

- TAG_BACKFILL_REQUESTS = "tag-backfill-requests"
- [x] Create on LaunchDarkly

**Affected issues**  
- ARC-1416

**How has this been tested?**  

**Whats Next?**
Roll the feature flag and celebrate 🥳

### CODE REVIEW

File: src/jira/client/jira-client.ts [hunk 1]
```diff
@@ -261,22 +262,37 @@ export const getJiraClient = async (
...
+					payload = {
+						builds: data.builds,
+						properties: {
+							gitHubInstallationId
+						},
+						providerMetadata: {
+							product: data.product
+						},
+						preventTransitions: options?.preventTransitions || false,
+						operationType: options?.operationType || "NORMAL"
```
Comment (tavomaciel) [inline src/jira/client/jira-client.ts:276]: Note: I'm adding `preventTransitions` on these too

File: src/config/feature-flags.ts [hunk 2]
```diff
@@ -90,3 +91,7 @@ export const isBlocked = async (installationId: number, logger: Logger): Promise
 		return false;
 	}
 };
+
+export const shouldTagBackfillRequests = async (): Promise<boolean> => {
+	return booleanFlag(BooleanFlags.TAG_BACKFILL_REQUESTS, false);
+};
```
Comment (tavomaciel) [inline src/config/feature-flags.ts:97]: I know you don't usually expose ffs as functions, but I thought this was a bit easier to mock on tests

File: src/sync/installation.ts [hunk 3]
```diff
@@ -296,14 +296,21 @@ async function doProcessInstallation(app, data: BackfillMessagePayload, sentry:
...
+							preventTransitions: true,
+							operationType: "BACKFILL"
+						});
 						break;
 					default:
 						await jiraClient.devinfo.repository.update(taskPayload.jiraPayload, {
-							preventTransitions: true
+							preventTransitions: true,
+							operationType: "BACKFILL"
 						});
```
Comment (tavomaciel) [inline src/sync/installation.ts:314]: From what I could find, this is where backfills requests happen. Please let me know if the options should be changed somewhere else.

Comment (joshkay10) [reply src/sync/installation.ts:314]: The above two jiraClient calls are for backfill too
`await jiraClient.workflow.submit(taskPayload.jiraPayload);`
`await jiraClient.deployment.submit(taskPayload.jiraPayload);`

**Review by gxueatlassian — APPROVED** [review]

File: docs/DevInfoAPI.json [hunk 4]
```diff
@@ -321,6 +321,16 @@
           "readOnly" : true,
           "default" : false
         },
+				"operationType": {
+					"type" : "string",
+					"description" : "",
```
Comment (daniel-hocking) [inline docs/DevInfoAPI.json:326]: Should this have a description / where is this being published?

File: src/config/feature-flags.ts (hunk 2 above)
Comment (mboudreau) [reply src/config/feature-flags.ts:97]: We don't want to have to maintain multiple functions.  Current way to mock is a pretty simple single line.  Plus, wouldn't you want to pass in the `jiraHost` for this?

File: src/interfaces/jira.ts [hunk 5]
```diff
@@ -190,3 +190,9 @@ export interface JiraRemoteLinkStatus {
 // https://atlassian.design/components/lozenge/examples
 export type JiraRemoteLinkStatusAppearance = "default" | "inprogress" | "moved" | "new" | "removed" | "prototype" | "success";
 
+export type JiraOperationType = "NORMAL" | "BACKFILL"
```
Comment (mboudreau) [inline src/interfaces/jira.ts:193]: not much point in exporting the type if it's not being used anywhere else. Could probably just inline it within `JiraSubmitOptions`

File: src/jira/client/jira-client.ts [hunk 6]
```diff
@@ -286,12 +302,24 @@ export const getJiraClient = async (
...
 					await subscription?.update({ syncWarning: issueKeyLimitWarning });
 				}
-				const payload = {
-					deployments: data.deployments,
-					properties: {
-						gitHubInstallationId
-					}
-				};
+				let payload;
+				if (await shouldTagBackfillRequests()) {
```
Comment (mboudreau) [inline src/jira/client/jira-client.ts:306]: same here

File: src/jira/client/jira-client.ts [hunk 7]
```diff
@@ -302,20 +330,32 @@ export const getJiraClient = async (
...
 					await subscription?.update({ syncWarning: issueKeyLimitWarning });
 				}
-				const payload = {
-					remoteLinks: data.remoteLinks,
-					properties: {
-						gitHubInstallationId
-					}
-				};
+				let payload;
+				if (await shouldTagBackfillRequests()) {
```
Comment (mboudreau) [inline src/jira/client/jira-client.ts:342]: and here

File: src/jira/client/jira-client.ts [hunk 8]
```diff
@@ -343,17 +383,30 @@ const batchedBulkUpdate = async (
...
 		}
-		const body = {
-			preventTransitions: options?.preventTransitions || false,
-			repositories: [data],
-			properties: {
-				installationId
-			}
-		};
+		let body;
+		if (shouldTagBackfillRequestsValue) {
```
Comment (mboudreau) [inline src/jira/client/jira-client.ts:392]: and here

**Pull request merged**